### PR TITLE
Initialize Azure Pipelines Agent Controller Definition

### PR DESCRIPTION
**Pull request by vmichel95**

### CODE REVIEW

File: ansible.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[defaults]
+roles_path = {{CWD}}/../
```
Comment (andrewrothstein) [inline ansible.cfg]: whatsdis?

File: defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
+# defaults file for azure_pipelines_agent-ctl
+azure_pipelines_agent_ctl_allow_root: True
+azure_pipelines_agent_ctl_install_dir: /usr/local/vsts-agent
+
+azure_pipelines_agent_ctl_install_dependencies: True
+azure_pipelines_agent_ctl_dependency_install_script: './bin/installdependencies.sh'
```
Comment (andrewrothstein) [inline defaults/main.yml]: in YaML no need to quote strings

File: meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  author: Victor Michel
+  description: Utility operations for managing Azure Pipelines Agent
+  company: BlackRock Financial Management
+  license: 
+    - MIT
+  min_ansible_version: 2.0
+  galaxy_tags: 
+    - development
+  platforms:
+  - name: EL
```
Comment (andrewrothstein) [inline meta/main.yml:13]: reduce to supported distros

File: vars/main.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+# vars file for azure_pipelines_agent-ctl
+azure_pipelines_agent_ctl_runas_root_env: '{{ ["AGENT_ALLOW_RUNASROOT", (azure_pipelines_agent_ctl_allow_root and "root" == azure_pipelines_agent_ctl_service_user) | bool | int] }}'
+azure_pipelines_agent_ctl_proxy_url_env: '{{ ["VSTS_HTTP_PROXY", azure_pipelines_agent_ctl_proxy["url"]]}}'
+azure_pipelines_agent_ctl_proxy_username_env: '{{ ["VSTS_HTTP_PROXY_USERNAME", azure_pipelines_agent_ctl_proxy["username"] | default("")] }}'
+azure_pipelines_agent_ctl_proxy_password_env: '{{ ["VSTS_HTTP_PROXY_PASSWORD", azure_pipelines_agent_ctl_proxy["password"] | default("")] }}'
+azure_pipelines_agent_ctl_proxy_env_vector: '{{ [azure_pipelines_agent_ctl_proxy_url_env, azure_pipelines_agent_ctl_proxy_username_env, azure_pipelines_agent_ctl_proxy_password_env] }}'
+azure_pipelines_agent_ctl_shell_env: '{{ dict([azure_pipelines_agent_ctl_runas_root_env] + azure_pipelines_agent_ctl_proxy_env_vector) | dict2items | rejectattr("value", "equalto", "") | list | items2dict }}'
+
+azure_pipelines_agent_ctl_base_flags: >-
+  --url {{ azure_pipelines_agent_ctl_devops_url }}
```
Comment (andrewrothstein) [inline vars/main.yml:11]: why does using ansible templates to build command lines feel so janky

**Comment by andrewrothstein** [comment]
having a '-' in a role name historically made the ansible-galaxy unhappy with you. I try to stick to _ in the role names. Careful when you import.

File: ansible.cfg (hunk 1 above)
Comment (vmichel95) [reply ansible.cfg]: Added for local role testing

**Pull request merged**